### PR TITLE
Convert production cassandra config to a template with volumeClaimTemplate support

### LINKED ISSUE #54: Add VolumeClaimTemplates for Cassandra
I'm not sure what the best approach is to include this, because it needs to be variablized. Specifically `storageClassName` and `resources.requests.storage`. I added the following under the StatefulSet `spec`:

```
    volumeClaimTemplates:
      - metadata:
          name: data
        spec:
          accessModes: [ "ReadWriteOnce" ]
          storageClassName: standard
          resources:
            requests:
              storage: 3Gi
      - metadata:
          name: logs
        spec:
          accessModes: [ "ReadWriteOnce" ]
          storageClassName: standard
          resources:
            requests:
              storage: 1Gi
```

### PR DESCRIPTION
**Pull request by gregswift**
## Which problem is this PR solving?
- Resolves #54 (Adding support for volumeClaimTemplates to the cassandra statefulset)
- Setups the StatefulSet to support scaling up the cluster with persistent volume claims
- Conversion to template is required to parameterize things like size and pool names for the PVC

## Short description of the changes
The built in cassadnra template doesnt really work with persistent data. Changing it to a template, and introducing volumeClaimTemplate in the StatefulSet makes it so its usable for being able to scaling at all.

That being said apparently according to #30 it doesnt seem like there is a goal of making the cassandra* bits in this repo prod ready?

### CODE REVIEW

**Comment by jpkrohling** [comment]
> That being said apparently according to #30 it doesnt seem like there is a goal of making the cassandra* bits in this repo prod ready?

That's correct: we provide the template as a starting point, but we have no intention in getting into the business of providing a production-grade Cassandra installation :-)

File: production/cassandra-template.yml [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        jaeger-infra: cassandra-replica
+    template:
+      metadata:
+        labels:
+          app: cassandra
+          jaeger-infra: cassandra-replica
+      spec:
+        terminationGracePeriodSeconds: 1800
+        containers:
+        - name: ${CASSANDRA_SERVICE_NAME
```
Comment (jpkrohling) [inline production/cassandra-template.yml]: Isn't it missing a `}`?

File: production/cassandra-template.yml [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      app: cassandra
+- apiVersion: "apps/v1beta1"
+  kind: StatefulSet
+  metadata:
+    name: ${CASSANDRA_SERVICE_NAME}
+    labels:
+      app: jaeger
+      jaeger-infra: cassandra-statefulset
+  spec:
+    serviceName: ${CASSANDRA_SERVICE_NAME
```
Comment (jpkrohling) [inline production/cassandra-template.yml]: Isn't it missing a `}`?

**Comment by jpkrohling** [comment]
Thanks for the PR! Looks good to me, except for a couple of places where it seems a `}` is missing.

**Comment by rackergs** [comment]
I thought i had caught all of those :/

updated.

my whole point in making this happen was to actually use this as the prod data store because it was already openshift prepared. So i'm a bit torn around what to move forward with :/

**Comment by jpkrohling** [comment]
> my whole point in making this happen was to actually use this as the prod data store because it was already openshift prepared. So i'm a bit torn around what to move forward with :/

You probably know this better than me, but having Cassandra on Kubernetes/OpenShift is tricky, as pods come and go without notice. As far as I understand, this is not a pattern that Cassandra is particularly fond of. @pavolloffay and/or @burmanm can probably give you more details, as they spent more time on it than I have. Besides, at the time we wrote this OpenShift template, there hasn't been any official/supported way of running Cassandra on them, other than a few [quick starts/tutorials](https://kubernetes.io/docs/tutorials/stateful-application/cassandra/) here and there. This might have changed, though.

In any case, the main point is that we expect users to understand what they are doing with their Cassandra installation and be able to self-support: we can help with the Jaeger parts, but we can't help with the Cassandra parts.

**Comment by jpkrohling** [comment]
Sorry for taking so long to merge this!

**Comment by gregswift** [comment]
thanks

**Pull request merged**